nx/lib/nx/binary_backend.ex

```diff
@@ -2033,6 +2051,36 @@ defmodule Nx.BinaryBackend do
     from_binary(t, new_data)
   end
 
+  @impl true
+  def triangular_solve(
+        %{shape: shape, type: output_type = b_holder},
+        %{type: input_type, shape: {rows, rows}} = a,
+        b
+      ) do
+    a_matrix = a |> to_binary() |> binary_to_matrix(input_type, {rows, rows})
+    b_vector = b |> to_binary() |> binary_to_matrix(input_type, {0, 1}) |> List.flatten()
+
+    solution = List.duplicate(0, rows)
+
+    Enum.with_index(a_matrix)
+    |> Enum.zip(b_vector)
+    |> Enum.reduce([], fn {current_row_and_idx, current_b}, previous_y ->
+      idx = elem(current_row_and_idx, 1)
+      row = elem(current_row_and_idx, 0)
+
+      y =
+        cond  do
+          idx == 0 -> current_b / Enum.at(row, idx)
+          idx in 1..rows - 1 -> current_b - Enum.at(row, idx) * Enum.at(previous_y, 0)
+          idx == rows - 1 -> current_b - Enum.at(row, idx) * Enum.sum(previous_y)
+        end
+
+      solution = replace_vector_element(solution, idx, y)
```

Comment:
```suggestion
      y = (current_b - dot_matrix(row, previous_y)) / Enum.at(row, idx)

      replace_vector_element(previous_y, idx, y)
```

Reply:
@polvalente This will surely help deal with all the cases.